Content.Server/Botany/Systems/BotanySwabSystem.cs

```diff
@@ -72,11 +86,49 @@ private void OnDoAfter(EntityUid uid, BotanySwabComponent swab, DoAfterEvent arg
             if (old == null)
                 return;
             plant.Seed = _mutationSystem.Cross(swab.SeedData, old); // Cross-pollenate
-            swab.SeedData = old; // Transfer old plant pollen to swab
+            if (swab.Contaminate == true)
+                swab.SeedData = old; // Transfer old plant pollen to swab
             _popupSystem.PopupEntity(Loc.GetString("botany-swab-to"), args.Args.Target.Value, args.Args.User);
         }
+        args.Handled = true;
+    }
 
+    private void OnClean(EntityUid uid, BotanySwabComponent swab, ref UseInHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (swab.Cleanable == true)
+        {
+            swab.SeedData = null;
+            _popupSystem.PopupClient(Loc.GetString("botany-swab-clean"), uid, args.User);
+            _audioSystem.PlayPvs(swab.CleanSound, uid);
+        }
         args.Handled = true;
```

Comment:
this stops any further actions
even if we did nothing
```suggestion
    private void OnClean(EntityUid uid, BotanySwabComponent swab, ref UseInHandEvent args)
    {
        if (args.Handled)
            return;

        if (!swab.Cleanable)
            return;
            
        swab.SeedData = null;
        _popupSystem.PopupClient(Loc.GetString("botany-swab-clean"), uid, args.User);
        _audioSystem.PlayPvs(swab.CleanSound, uid);
        args.Handled = true;
```